stack.py

```diff
@@ -1,49 +1,99 @@
-# Make Python 2 use float division instead of integer division
+# -*- coding: utf-8 -*-
 from __future__ import division
+import re
 
 
-# TODO: This is pretty stringly typed, perhaps return a custom object?
-def next_instruction(program):
-    split_program = program.split(";")
-    for instuction in split_program:
-        if instuction == "":
-            raise StopIteration
+class Instr(object):
+    sigil_to_op = {
```

Comment:
`dup` is missing from `sigil_to_op`. Perhaps U+21A4 (↤) or U+21E4 (⇤) can represent it.